tls/s2n_x509_validator.c

```diff
@@ -503,9 +510,13 @@ static S2N_RESULT s2n_x509_validator_verify_cert_chain(struct s2n_x509_validator
 
     uint64_t current_sys_time = 0;
     RESULT_GUARD(s2n_config_wall_clock(conn->config, &current_sys_time));
+    if (sizeof(time_t) == 4) {
```

Comment:
personally, I actually like the more explicit logic, because it is _very_ clear what exactly it means. But I've also spent a lot of time staring at this, so if you think it'd be nicer with the `s2n_supports_large_time_t()` I'm happy to make the change.

Reply:
No, that makes sense!